frame/nomination-pools/src/lib.rs

```diff
@@ -1300,9 +1327,12 @@ impl<T: Config> RewardPool<T> {
 		// x = 100
 		//
 		// which is basically 10^-8 DOTs. See `smallest_claimable_reward` for an example of this.
-		T::RewardCounter::checked_from_rational(payouts_since_last_record, bonded_points)
-			.and_then(|ref r| self.last_recorded_reward_counter.checked_add(r))
-			.ok_or(Error::<T>::OverflowRisk)
+		let current_reward_counter =
+			T::RewardCounter::checked_from_rational(share_rewards, bonded_points)
+				.and_then(|ref r| self.last_recorded_reward_counter.checked_add(r))
+				.ok_or(Error::<T>::OverflowRisk)?;
+
+		Ok((current_reward_counter, share_commission))
```

Comment:
`share_commission` is not being subject to `bonded_points`, i am not sure if this is causing errors.